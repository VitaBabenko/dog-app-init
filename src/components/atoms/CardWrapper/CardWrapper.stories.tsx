import type { Meta, StoryObj } from '@storybook/react';
import { CardWrapper } from '.';

const meta = {
  title: 'Atoms/CardWrapper',
  component: CardWrapper,
  argTypes: {
    variant: {
      options: ['primary', 'smallRadiusBorder', 'bigRadiusBorder'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof CardWrapper>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    variant: 'primary',
    borderRadius: 20
  }
};

export const SmallRadiusBorder: Story = {
  args: {
    variant: 'smallRadiusBorder',
    borderRadius: 10
  }
};

export const BigRadiusBorder: Story = {
  args: {
    variant: 'bigRadiusBorder',
    borderRadius: 40
  }
};
