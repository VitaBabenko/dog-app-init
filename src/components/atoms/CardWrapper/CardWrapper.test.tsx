import { composeStories } from '@storybook/react';
import * as stories from './CardWrapper.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('CardWrapper', () => {
  const { Primary, SmallRadiusBorder, BigRadiusBorder } =
    composeStories(stories);
  testSnapshot(<Primary />);
  testSnapshot(<SmallRadiusBorder />);
  testSnapshot(<BigRadiusBorder />);
});
