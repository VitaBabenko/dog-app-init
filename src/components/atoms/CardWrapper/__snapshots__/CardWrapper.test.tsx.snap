// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWrapper should match snapshot Component 1`] = `
.emotion-0 {
  border-radius: 80px;
  position: relative;
  cursor: pointer;
  -webkit-background-position: center;
  background-position: center;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  overflow: hidden;
  max-width: 590px;
  min-height: 90px;
  background-color: #FFF;
  box-shadow: 6px 6px 0px 0px #000000;
  border: 1px solid #090808;
  border-radius: 20px;
}

.emotion-0 img {
  display: block;
  width: 100%;
  object-fit: cover;
}

.emotion-0:hover,
.emotion-0:active {
  box-shadow: 6px 6px 8px 0px #921FED;
}

<div>
  <div
    class="MuiBox-root emotion-0"
    variant="primary"
  />
</div>
`;

exports[`CardWrapper should match snapshot Component 2`] = `
.emotion-0 {
  border-radius: 40px;
  position: relative;
  cursor: pointer;
  -webkit-background-position: center;
  background-position: center;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  overflow: hidden;
  max-width: 590px;
  min-height: 90px;
  background-color: #FFF;
  box-shadow: 3px 3px 0px 0px #000000;
  border: unset;
  border-radius: 10px;
}

.emotion-0 img {
  display: block;
  width: 100%;
  object-fit: cover;
}

.emotion-0:hover,
.emotion-0:active {
  box-shadow: 3px 3px 0px 0px #000000;
}

<div>
  <div
    class="MuiBox-root emotion-0"
    variant="smallRadiusBorder"
  />
</div>
`;

exports[`CardWrapper should match snapshot Component 3`] = `
.emotion-0 {
  border-radius: 160px;
  position: relative;
  cursor: pointer;
  -webkit-background-position: center;
  background-position: center;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  overflow: hidden;
  max-width: 590px;
  min-height: 90px;
  background-color: #FFF;
  box-shadow: 6px 6px 0px 0px #000000;
  border: 1px solid #090808;
  border-radius: 40px;
}

.emotion-0 img {
  display: block;
  width: 100%;
  object-fit: cover;
}

.emotion-0:hover,
.emotion-0:active {
  box-shadow: 6px 6px 8px 0px #921FED;
}

<div>
  <div
    class="MuiBox-root emotion-0"
    variant="bigRadiusBorder"
  />
</div>
`;
